src/Symfony/Bundle/SecurityBundle/Resources/config/password_hasher.php

```diff
@@ -27,4 +32,22 @@
         ->alias('security.password_hasher', 'security.user_password_hasher')
         ->alias(UserPasswordHasherInterface::class, 'security.password_hasher')
     ;
+
+    if (class_exists(PasswordHasherExtension::class)) {
```

Comment:
We must not use conditionals in the service configuration file. We should either move these definitions to a new config file and conditionally load it, or always register the services and conditionally remove the definitions from the DI extension (preferred solution).

See e.g. https://github.com/symfony/symfony/blob/8a5b9a27250736a9b38e2bd802c394a2834e7e94/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L121-L125

Reply:
Oh, good suggestion. Yeah, let's move this to FrameworkBundle (as the PasswordHasher component is separated from the Security one).

Reply:
@wouterj I have updated my PR to remove the condition in the config file.
I finally did not move the config to the `FrameworkBundle`, because this feature rely on [`PasswordAuthenticatedUserInterface`](https://github.com/symfony/symfony/blob/ef8b11061cd0cb7c1950c9783090b93cb169ae54/src/Symfony/Component/Form/Extension/PasswordHasher/EventListener/PasswordHasherListener.php#L53-L55) which is part of `security-core`, not only on the `password-hasher` component.

Reply:
@wouterj any chance you can review your requested changes? Thank you